### PR TITLE
fix: recurring donation dates

### PR DESCRIPTION
**Pull request by Shehryar21**
## Implementation description. How did you make this change?
* Changed recurring donation end date to be set as end of day in the frontend. Currently recurring end date is set to be the date that is selected in the date picker and the time is just the current time, but with the change I made, time is now end of day. The old timing messed up how donations are created in the backend and sometimes one less schedule was created because of this. Take a look at the following screenshot: 
image
A recurring donation should have been added for Mar 26 but it did not.

Now you might be thinking why could I not do this change in the backend function instead of frontend?
This is because when we use dayjs in the backend, it converts the time into UTC. This creates some edge cases bugs, for example lets say recurring donation end date was Mar 27 10pm, with dayjs in backend this will become Mar 28 2am (since UTC is 4 hours ahead of EST), then if I set the date to EOD as I did in frontend right now, recurring end date will become Mar 28 11:59 am which will then create an extra donation cuz of how our while loop and recurring donation logic is implemented.

Why doesnt the UTC conversion happen in the frontend?
From my understanding, in the `SelectDateTime.tsx`, we are using Javascript dates and timezone is preserved so it is not converted to UTC

Lol this is very hard to explain in text, but if someone is confused with this, lmk and I can explain in more depth but I am hoping this can fix the recurring donation dates saga once and for all (fingers crossed).

## Steps to test
1. Create different daily recurring donations with different times and edge cases (4am, 12am, 8pm, etc) and ensure accurate number of daily donations are being created. If end date is Mar 26 for example, the last donation for that should be on Mar 26 as well.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] The appropriate tests if necessary have been written

### CODE REVIEW

File: backend/typescript/services/implementations/schedulingService.ts [hunk 1]
```diff
@@ -525,7 +525,6 @@ class SchedulingService implements ISchedulingService {
         const recurringDonationEndDate = dayjs(
           scheduling.recurringDonationEndDate!,
         );
-        recurringDonationEndDate.startOf("day");
```
Comment (Shehryar21) [inline backend/typescript/services/implementations/schedulingService.ts:528]: This line was actually useless since this returns a dayjs object `recurringDonationEndDate` was never being set to start of day. It should have been `recurringDonationEndDate = recurringDonationEndDate.startOf("day")`. But it doesnt matter cuz we dont need it anymore.

File: frontend/src/components/pages/Scheduling/SelectDateTime.tsx [hunk 2]
```diff
@@ -249,6 +249,7 @@ const SelectDateTime = ({
     setRecurringEndDate(selectedDateObj);
 
     const recurringDate = new Date(selectedDateObj);
+    recurringDate.setHours(23,59,59,999); // Need to set it to EOD so that recurring schedules are created properly in the backend
```
Comment (hanlinc27) [inline frontend/src/components/pages/Scheduling/SelectDateTime.tsx:252]: let's use this? (https://date-fns.org/v2.28.0/docs/endOfDay)
```suggestion
    const recurringDate = endOfDay(new Date(selectedDateObj))
```

**Review by hanlinc27 — CHANGES_REQUESTED** [review]

**Review by hanlinc27 — APPROVED** [review]
